### PR TITLE
tangent arcs

### PR DESCRIPTION
**Pull request by marcus7070**
I'm working on added 2D tangent arcs. 

Still to do:

- [x] Tests for `occ_impl.shapes.Edge.makeTangentArc`
- [x] Tests for `Workplane._findFromEdge`:
  - [x] Test it passes with an edge
  - [x] Test it fails with other objects on the stack
- [x] Docs
- [ ] ~Consider adding `CQ.tangentArcWithRadius`~
- [x] Don't forget to run the Black code linting this time!

### CODE REVIEW

**Comment by marcus7070** [comment]
Some points for the reviewers to look at:

- In Edge.makeTangentArc I access `v2._wrapped` to get at the gp_vec that is wrapped by Vector. Seems like the best way to do it, but I still feel weird directly accessing an underscored attribute.
- I'm open to changing the names of these methods. In particular, I note that `tangentArcTo` uses coords relative to the current point (which I found to be the most common use case when I was testing this method), while methods like `lineTo`, `hLineTo`, `moveTo`, and `polarLineTo` all use coords relative to the workplane. Actually, now I write that out I realise I have to change the name. Anyone got any suggestions? I'd prefer something more descriptive than `tangentArc` because I expect at some point I'll add the method `tangentArcRadius`, which will create an arc with arguments for radius and angle, and I know I will confuse those two names.
- I find the docstring documentation sufficient, but I can add an example if anyone thinks it necessary. 
- OCC is capable of making tangent arcs to points in three dimensions. Which is pretty impressive when you see it in action. I couldn't think of any way to incorporate that ability into the fluent API though.
- If you were to remove the line `endpoint = self.plane.toWorldCoords(endpoint)` from `Workplane.tangentArcTo`, all my tests would still pass but in actual use your tangentArcs would be all over the place. Perhaps I should write another test?

I also fixed a typo in examples.rst that was causing an error while building docs.

**Comment by marcus7070** [comment]
damn it forgot black checks again!

**Comment by marcus7070** [comment]
Going to rename `tangentArcTo` to `tangentArcEndpoint`. I'm still open to better names.

**Comment by marcus7070** [comment]
Rebased on master.

**Comment by jmwright** [comment]
@marcus7070 Are you wanting this to be reviewed now?

**Comment by marcus7070** [comment]
@jmwright yes please!

**Comment by marcus7070** [comment]
whoops, forgot to change the docs when I renamed the method, just a second

File: doc/apireference.rst [hunk 1]
```diff
@@ -54,6 +54,7 @@ All 2-d operations require a **Workplane** object to be created.
 	Workplane.threePointArc
 	Workplane.sagittaArc
 	Workplane.radiusArc
+  Workplane.tangentArcTo
```
Comment (marcus7070) [inline doc/apireference.rst]: Should be `Workplane.tangentArcEndpoint`

Comment (jmwright) [reply doc/apireference.rst]: @marcus7070 I would go ahead and make the change.

**Comment by jmwright** [comment]
@marcus7070 I get the following deprecation warnings when running with pytest 5.3.2. Might as well change them now, as long as there's not some weird backwards compatibility issue. Other than that, I think this PR looks good. Thanks for the contribution. I'll wait for feedback from @adam-urbanczyk on this before I merge.
```
tests/test_cadquery.py::TestCadQuery::testTangentArcToPoint
  /home/jwright/Downloads/repos/cadquery/tests/test_cadquery.py:3070: DeprecationWarning: Please use assertAlmostEqual instead.
    self.assertAlmostEquals(area0, (1 + math.pi * 0.5 ** 2), 4)

tests/test_cadquery.py::TestCadQuery::testTangentArcToPoint
  /home/jwright/Downloads/repos/cadquery/tests/test_cadquery.py:3085: DeprecationWarning: Please use assertAlmostEqual instead.
    self.assertAlmostEquals(s1.val().Volume(), s0.val().Volume(), 4)

tests/test_cadquery.py::TestCadQuery::testTangentArcToPoint
  /home/jwright/Downloads/repos/cadquery/tests/test_cadquery.py:3099: DeprecationWarning: Please use assertAlmostEqual instead.
    s1.faces(">Z").val().Area(), 2 * 3 + 0.5 * math.pi * 0.5 ** 2, 4

tests/test_cadquery.py::TestCadQuery::testTangentArcToPoint
  /home/jwright/Downloads/repos/cadquery/tests/test_cadquery.py:3116: DeprecationWarning: Please use assertAlmostEqual instead.
    self.assertAlmostEquals(s2.val().Volume(), math.pi, 1)
``

**Comment by marcus7070** [comment]
Made the change in the tests, thanks @jmwright!

**Comment by adam-urbanczyk** [comment]
I'll try to take a look today @jmwright

File: cadquery/cq.py [hunk 2]
```diff
@@ -1722,6 +1744,37 @@ def radiusArc(self, endPoint, radius, forConstruction=False):
         else:
             return self.sagittaArc(endPoint, -sag, forConstruction)
 
+    def tangentArcEndpoint(self, endpoint, forConstruction=False, relative=True):
```
Comment (adam-urbanczyk) [inline cadquery/cq.py]: Regarding the name, why not just `tangentArcPoint`? Slightly shorter and I think descriptive enough.

File: cadquery/cq.py [hunk 3]
```diff
@@ -1722,6 +1744,37 @@ def radiusArc(self, endPoint, radius, forConstruction=False):
...
+        """
+        Draw an arc as a tangent from the end of the current edge to endpoint.
+
+        :param endpoint: point for the arc to end at
+        :type endpoint: 2-tuple or Vector
```
Comment (adam-urbanczyk) [inline cadquery/cq.py]: Will it not work with 3-tuple as well? Maybe not the most common use case but I can think of some applications.

Comment (marcus7070) [reply cadquery/cq.py]: It should work just fine in it's current form with 3-tuples, creating 3D arcs. I just didn't write any tests for it so I didn't want to advertise that in the docstring.

Comment (marcus7070) [reply cadquery/cq.py]: Docstring now mentions 3-tuples.

**Comment by marcus7070** [comment]
I haven't written any tests for `tangentArcPoint` with a 3-tuple argument. Feel free to merge without those tests or else I'll write them when I have some time (won't be for a few days at least).

**Comment by adam-urbanczyk** [comment]
I'll try to add a case myself then.

**Comment by adam-urbanczyk** [comment]
Looks good I think, feel free to merge @jmwright  .

**Comment by jmwright** [comment]
@marcus7070 Thanks for this contribution.

@adam-urbanczyk Thanks for the feedback and the additional test.

**Pull request merged**